Decisions/scs-XXXX-v1-requirements-for-sso-identity-federation.md

```diff
@@ -0,0 +1,196 @@
+to avoid the need to frequently extend/reduce client service configuration
+whenever a new IdP federation needs to be added to Keycloak to onboard
+a new customer. This is relevant for services like OpenStack Keystone,
+which currently cannot be easily reconfigured for new SSO endpoints
+without restarting the service, making the service unavailable for
+a short span of time and increasing risk connected with service restarts.
+
+Since version 17, Keycloak claims that it's capability for
+"cloud native" deployments on Kubernetes has improved.
+
+[Keycloak is offering a REST API](https://www.keycloak.org/docs-api/20.0.1/rest-api/index.html)
+for all aspects of its administration interface.
+
+For storage of Keycloak configuration and local user metadata
+(e.g. from which external IdP a user account originally came from)
+Keycloak supports several SQL backends through JDBC. Thus
+it can be hooked up to a Postgres Database or to a
+MariaDB/Galera cluster e.g..
+
+### Zitadel
+
+Zitadel is a newer implementation of a SSO IdP. It is implemented
+in Go and under active development.
+
+It came to attention of the SCS project because it offers a
+fresh take of an organization focussed data model, which has
+the potential to simplify IdP federation to SCS customer domains
+in the following areas:
+* For client services (single set of HTTP API endpoints).
+* For SCS operators for provisioning customer [organizations](https://zitadel.com/docs/concepts/structure/organizations)
+  and robust configuraton by using templated client, role and mapping
+  configuration.
+* For SCS customers for a robust user experience for self servicing.
+
+[Zitadel is offering REST APIs](https://zitadel.com/docs/apis/introduction)
+for multiple areas of use and configuration.
+
+In comparison to Keycloak it currently lacks support for the
+[Device Authorization Grant](https://github.com/SovereignCloudStack/issues/issues/221),
+which, at time of writing, is a feauture that is relevant
+for SCS to be able use OpenStack CLI and APIs with federated
+identities. At Zitadel there are at least two [Github issues](https://github.com/zitadel/oidc/issues/141)
+proposed to implement this feature.
+
+At the time of writing it lacks support to consume identity data from LDAP backends.
+But that is currently not a relevant factor in the SCS context.
+
+At time of writing an PoC "spike" is done to assess and verify the hopes
+connected with Zitadel in the context of the SCS testbed.
+
+# Open questions
+
+* How would we implement testbed deployment support for Zitadel?
+    * e.g. `wsgi-keystone.conf` would need to look different. One template covering both options?
+    * e.g. steps like `openstack federation protocol create` would probably be different.
+* Should we support both as options?
+    * What's the benefit?
+    * How would we allow SCS operators to choose?
+* Do we need some kind of SWOT analysis to come to a decision?
+
+# Decision
+
+_Decision_
+
```

Comment:
As this document gathers the requirements, I’d remove the Decisions part and add it in a subsequent version of this document, if we actually took decisions.